### PR TITLE
Icon styling [DON'T DEPLOY UNTIL 13/12]

### LINKED ISSUE #10746: Icon styling on the Fill component (waiting for cartodb-platform/issues/2764) [DON'T DEPLOY UNTIL 12/13]
This feature enables the styling with icons in Builder through the Fill component.

Design: https://github.com/CartoDB/cartodb/issues/10452

PR (and information on how to test it): https://github.com/CartoDB/cartodb/pull/11005

### Implemented
- [x] Add actual icon in the input field (#10852)
- [x] Create flag for the feature (#10847) (`icon-styling`)
- [x] Use an image and a color on the img/color row.
- [x] Add image pane inside the Solid Pane
- [x] Add Img / color row 
- [x] Update CartoCSS.
- [x] Added basic image pane
- [x] Show assets in the modal.
- [x] Add modal to select the assets.
- [x] Create the assets collection to retrieve the assets from the backend
- [x] Add/fix specs (https://github.com/CartoDB/cartodb/issues/10875).
- [x] Add styles in the modal (@piensaenpixel)
- [x] Enable CORS for the assets bucket (https://github.com/CartoDB/cartodb-platform/issues/2764)

### Future features

- [ ] Add upload file system.
- [ ] Show patterns.
- [ ] Add [IMG] icon to the input fill.

### PR DESCRIPTION
**Pull request by javierarce**
This PR enables icon styling through the fill component. 

This feature is hidden from users using the `icon-styling` feature flag.

Although there are many changes in several files, most of the changes are simply passing the `modals` and `userModel` objects to the fill component, so we can launch a modal and check the feature flag inside it.

### Pending stuff

In order to speed up the development of this feature, and since it's protected under a feature flag, there are some minor details that this PR is not addressing at this moment (they'll be taken care on separate issues):

1. Small sets of specs (they cover the bare minimum, but we'll add some more in another ticket)
2. The color of the icon in the input UI gets lost after the first change of color.
3. CSS misalignment in the icon carousel.

### How to test this feature

1. Enable the `icon-styling` feature flag in your user.
2. Create a map with some geometries.
3. Go to the style tab.
4. Change the `fill` property of the geometries using the fill component.
5. Select an icon from the list or the modal.

Once you've done that, your cartoCSS should be updated with a `marker-file` property. Please, also test that disabling the feature flag hides the UI from the user.

You can have a look the UI in this issue: https://github.com/CartoDB/cartodb/issues/10452 and on [this one](https://github.com/CartoDB/design/issues/303), but bear in mind that we are not implementing file upload or styling categories just yet. 

### Deploying the feature

We'll deploy this feature and enable it for our own CARTO users first. In a next phase we'll add more  functionality and make it available for the general public. Ask @xavijam or @javierarce if you have any doubt with this.

Closes #10746

### CODE REVIEW

**Comment by xavijam** [comment]
Do we need shwrinkwrap changes in this PR?

**Comment by javierarce** [comment]
My change just updates the version:

from `"version": "4.5.48"` to `"version": "4.5.56"`

File: lib/assets/test/spec/cartodb3/components/form-components/editors/fill/input-color-fixed-content-view.spec.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (MariaCheca) [inline lib/assets/test/spec/cartodb3/components/form-components/editors/fill/input-color-fixed-content-view.spec.js]: Hey, what happened to this tests? Can't it be tested yet (something's left from another PR) or is it a mistake? :)

**Comment by MariaCheca** [comment]
I've tested it in staging and it looks ok, works well ✨ 

The only thing is that bottom line when the color is selected looks one pixel above its right position:
screen shot 2016-12-12 at 16 40 52

**Comment by javierarce** [comment]
@MariaCheca good catch!… don't worry about that issue at this moment, we'll fix it in another issue.

**Comment by xavijam** [comment]
I'd add that thing with the other problem with the scroll, like small details to be fixed.

**Comment by MariaCheca** [comment]
Will deploy tomorrow, it works perfectly :)

**Pull request merged**